shoop/core/taxing/utils.py

```diff
@@ -6,13 +6,53 @@
 # This source code is licensed under the AGPLv3 license found in the
 # LICENSE file in the root directory of this source tree.
 
+from collections import defaultdict
+
 from shoop.core.pricing import TaxfulPrice, TaxlessPrice
 from shoop.utils.money import Money
 
 from ._line_tax import SourceLineTax
 from ._price import TaxedPrice
 
 
+def get_tax_class_proportions(lines):
+    """
+    Generate tax class proportions from taxed lines.
+
+    Sum prices per tax class and return a list of (tax_class, factor)
+    pairs, where factor is the proportion of total price of lines with
+    given tax class from the total price of all lines.
+
+    :type lines: list[shoop.core.order_creator.SourceLine]
+    :param lines: List of taxed lines to generate proportions from
+
```

Comment:
Actually it was intentional, to separate input params from return param, but it seems that other docstrings don't have a line break there, so I'll remove this too. :)